### PR TITLE
Lf 3686 update tiles component to new design with description and checkbox

### PR DESCRIPTION
**Pull request by Duncan-Brain**
**Description**

This PR adds components for the new default and custom expense and revenue types.

The Jira ticket suggest adding onto the `<Tiles />` component but as I was adapting it to accommodate a "layout" prop and different styling I thought I shouldn't mess with the definition of Tiles as simple with not too much information, and in a gallery or matrix layout. 

I chose List as the component group name and accept the similarity to tiles as an acceptable duplication as it was almost entirely copied from tiles. 

Two List Item options are available here: 
- IconDescriptionCheckbox - has a selected prop for a fake checkbox, it is expected that onClick() it will change the checkbox image to look checked
- IconDescription - removes the checkbox and selected, it is expected that onClick() will link to a new page and provide some data to the next screen

Stories:
[IconDescriptionCheckbox](http://localhost:6006/?path=/docs/components-list-icondescriptioncheckboxlistitem--docs)
[IconDescription](http://localhost:6006/?path=/docs/components-list-icondescriptionlistitem--docs)
[Revenue and Expense example Lists](http://localhost:6006/?path=/docs/components-list-list--docs)

Jira link: [LF-3686](https://lite-farm.atlassian.net/browse/LF-3686)

**Type of change**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Passes test case
- [x] UI components visually reviewed on desktop view
- [x] UI components visually reviewed on mobile view
- [ ] Other (please explain)

**Checklist:**

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] The precommit and linting ran successfully
- [ ] I have added or updated language tags for text that's part of the UI
- [ ] I have added "MISSING" for all new language tags to languages I don't speak
- [ ] I have added [the GNU General Public License](https://lite-farm.atlassian.net/l/cp/BT0Dd7WW) to all new files

[LF-3686]: https://lite-farm.atlassian.net/browse/LF-3686?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by kathyavini** [comment]
It's truly a thing of beauty!!! 😍 

I think the ticket might need to also include slotting it into the actual Expense and Revenue views, though. There isn't a separate task ticket for that -- I'm assuming because this one was written assuming `<Tiles />` would be used, and so that update would have happened automatically.

**Comment by Duncan-Brain** [comment]
Thanks @kathyavini there is a linked issue on the ticket I thought was in charge of the integration part.

**Comment by kathyavini** [comment]
@Duncan-Brain Ah yes the linked ticket does have a misleading name. But it's one of those placeholder user story tickets that aren't actually assigned to anyone.

**Comment by Duncan-Brain** [comment]
@kathyavini It doesn't have [PLACEHOLDER] but I agree it seems like an oversight to not assign it. It also has a search bar and filter shown on that ticket which I don't see other tickets for so I think its a real ticket that is testable. 

I can add the stuff if everyone thinks thats best but its going to double the files touched and increase the scope of work to include custom expense types, revenue types, custom revenue types. Maybe thats why your planning poker estimate was higher than the others haha.

**Comment by kathyavini** [comment]
HA! That would make sense on the planning poker 😆 Okay let's clarify in standup and get the other ticket assigned if necessary. I thought search was dropped from scope, but seeing how long this list is on mobile I wouldn't mind trying to get it in alongside the transaction list search.

File: packages/webapp/src/components/List/ListItems/IconDescriptionCheckbox/styles.module.scss [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  min-height: 64px;
+  padding: 0 4px;
+  overflow: hidden;
+  cursor: pointer;
+}
+
+.icon {
+  width: 36px;
+  height: 36px;
+  margin: 18px 9px;
```
Comment (antsgar) [inline packages/webapp/src/components/List/ListItems/IconDescriptionCheckbox/styles.module.scss:34]: Super minor thing, but Loïc has mentioned that he tries to maintain an 8px grid in his designs. I think in this case the spacing is a bit weird because the icons are different sizes, but when in doubt I'd suggest sticking to multiples of 8, so in this case 16 and 4 or 16 and 8.

File: packages/webapp/src/components/List/ListItems/IconDescriptionCheckbox/styles.module.scss [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+@media (hover: hover) {
+  .listItem:hover {
+    background-color: var(--teal100);
+  }
+}
+
+.content {
+  width: 100%;
+  margin: 9px 0px;
```
Comment (antsgar) [inline packages/webapp/src/components/List/ListItems/IconDescriptionCheckbox/styles.module.scss:56]: Same here, 8px margin although this is definitely a nitpicky and non blocking.

File: packages/webapp/src/components/List/index.jsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
+/*
```
Comment (antsgar) [inline packages/webapp/src/components/List/index.jsx:1]: `List` sounds a bit generic to me in the sense that I could imagine us having other kinds of list components in the future. Should we maybe call this something like `TileList`, or similar?

**Review by antsgar — COMMENTED** [review]
Left a couple of small comments, but the new component looks 🔥 nicely done!! 🎉

File: packages/webapp/src/stories/List/List.stories.jsx [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+import React from 'react';
+import { expect } from '@storybook/jest';
+import { within } from '@storybook/testing-library';
+import { componentDecorators } from '../Pages/config/Decorators';
+import List from '../../components/List';
+import { listItemTypes } from '../../components/List/constants';
+import { ReactComponent as EquipIcon } from '../../assets/images/log/equipment.svg';
```
[the remark this answers is not in the document]
Comment (kathyavini) [reply packages/webapp/src/stories/List/List.stories.jsx:21]: You could also:

```
import { icons as revenueIcons } from '../../containers/Finances/AddSale/RevenueTypes';
import { icons as expenseIcons } from '../../containers/Finances/NewExpense/ExpenseCategories';

const icons = {
  ...revenueIcons,
  ...expenseIcons,
};
```

as Sayaka exported those objects for use elsewhere.

I'm not sure how the pro of keeping the Storybook independent of the container weighs up against the con of having to keep the icon imports synched up in two places, though

File: packages/webapp/src/components/List/ListItems/IconDescriptionCheckbox/IconDescriptionCheckboxListItem.jsx [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+      onClick={onClick}
+      className={clsx(className, styles.listItem, selected && styles.listItem__selected)}
+      {...props}
+    >
+      <div className={styles.icon}>{icon}</div>
+      <div className={styles.content}>
+        <div className={styles.content_label}>{label}</div>
+        {description && <div className={styles.content_description}>{description}</div>}
+      </div>
+      {selected ? (
```
[the remark this answers is not in the document]
Comment (kathyavini) [reply packages/webapp/src/components/List/ListItems/IconDescriptionCheckbox/IconDescriptionCheckboxListItem.jsx:44]: I can't quite logic it all the way out, but could there even be a easy way to include Loïc's dream of a chevron in one of the views?

File: packages/webapp/src/stories/List/List.stories.jsx [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+import React from 'react';
+import { expect } from '@storybook/jest';
```
Comment (kathyavini) [inline packages/webapp/src/stories/List/List.stories.jsx:16]: These two imports not needed if not adding the interaction tests

**Review by kathyavini — DISMISSED** [review]
I was going to use the exact same emoji as @antsgar did! It looks absolutely 🔥!!!

I liked that you went with preserving current Tiles but keeping the similar structure for this one -- I tried it in all the views and it's really easy to switch over.

IF the chevron thing is a possibility, great... otherwise no worries, I think the manage custom types can be blank tiles.

P.S. +1 to `<TileList />`!! That's what I've always called that new view in my mind 🙂

File: packages/webapp/src/components/List/index.jsx (hunk 3 above)
Comment (Duncan-Brain) [reply packages/webapp/src/components/List/index.jsx:1]: But wouldn't other list components be described as a different listItems? List is meant to be very generic in the sense it is just a row that takes up 100% width --there is almost no styling on this component. A tile would be added to the tile types and have a matrix formation.

File: packages/webapp/src/components/List/ListItems/IconDescriptionCheckbox/styles.module.scss (hunk 2 above)
Comment (Duncan-Brain) [reply packages/webapp/src/components/List/ListItems/IconDescriptionCheckbox/styles.module.scss:56]: Ah okay I can do that -- I just was going off of the figma numbers I saw

File: packages/webapp/src/components/List/ListItems/IconDescriptionCheckbox/styles.module.scss (hunk 1 above)
Comment (Duncan-Brain) [reply packages/webapp/src/components/List/ListItems/IconDescriptionCheckbox/styles.module.scss:34]: Yeah good to know -- I saw 9 and 18 in figma. I can fix that up.

File: packages/webapp/src/components/List/ListItems/IconDescriptionCheckbox/IconDescriptionCheckboxListItem.jsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (Duncan-Brain) [reply packages/webapp/src/components/List/ListItems/IconDescriptionCheckbox/IconDescriptionCheckboxListItem.jsx:44]: I considered this.. it started to look messy on the parent List but let me try again. My other thought was the other would just be deleted once crop based expenses were added anyways so it would be adding logic that would just be deleted. But with chevron.. that makes more sense.

File: packages/webapp/src/stories/List/List.stories.jsx (hunk 6 above)
Comment (Duncan-Brain) [reply packages/webapp/src/stories/List/List.stories.jsx:16]: nice good find!

File: packages/webapp/src/stories/List/List.stories.jsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (Duncan-Brain) [reply packages/webapp/src/stories/List/List.stories.jsx:21]: Yeah I considered that actually, I guess for some reason it seemed weird -- but its not thaat weird. I can add that in.

**Comment by Duncan-Brain** [comment]
@kathyavini && @antsgar  

I have added the suggested changes except for the name change -- my understanding is that tiles are in a gallery or matrix layout and have very limited information on them (a description would maybe be too much).

List was meant to be very generic style wise so that other list items could handle the variation

**Review by kathyavini — APPROVED** [review]
Chevron!! 👌

File: packages/webapp/src/stories/List/IconDescriptionCheckboxListItem.stories.jsx [hunk 7]
```diff
@@ -0,0 +1,75 @@
+/*
```
Comment (kathyavini) [inline packages/webapp/src/stories/List/IconDescriptionCheckboxListItem.stories.jsx:1]: Are there three Storybook files even though it's one component so you can show the text variants for each one individually (rather than 9 variants under one giant file)? Just curious, not a blocker for merge or anything.

I don't know Storybook organization well enough to know if there is a more concise way to show all those combinations.

File: packages/webapp/src/components/List/index.jsx (hunk 3 above)
Comment (antsgar) [reply packages/webapp/src/components/List/index.jsx:1]: Ah yeah, that's a good point! I'm on board 🙂

File: packages/webapp/src/stories/List/IconDescriptionCheckboxListItem.stories.jsx (hunk 7 above)
Comment (Duncan-Brain) [reply packages/webapp/src/stories/List/IconDescriptionCheckboxListItem.stories.jsx:1]: Basically yeah, I am not sure best practice either but it seemed like a good idea to have a Story for each as they will be used different places throughout the app. 🤷 could combine it if that makes sense to people but yeah.

**Review by antsgar — APPROVED** [review]
